src/xdocs/config_whitespace.xml

```diff
@@ -600,21 +600,68 @@ class Test {
       </subsection>
 
       <subsection name="Examples" id="FileTabCharacter_Examples">
-        <p>
-          To configure the check to report on the first instance in each
-          file:
-        </p>
+        <p>To configure the check to report only the first instance in each file:</p>
         <source>
 &lt;module name=&quot;FileTabCharacter&quot;/&gt;
         </source>
-        <p>
-          To configure the check to report on each line in each file:
-        </p>
+        <p>Example:</p>
+        <source>
+public class test {
+  int a;                     // violation : indented using tab, no further checks
+
+  public void foo (int arg) {
+      a = arg;
+  }
+}
+        </source>
+        <p>To configure the check to report each instance in each file:</p>
         <source>
 &lt;module name=&quot;FileTabCharacter&quot;&gt;
   &lt;property name=&quot;eachLine&quot; value=&quot;true&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <p>Example:</p>
+        <source>
+public class test {
+  int a;                            // violation : indented using tab
+
+  public void foo (int arg) {       // violation : indented using tab
+      a = arg;                      // OK, indented using spaces
+  }                                 // OK, indented using spaces
+}
+        </source>
+        <p>To configure the check to report instances on only certain file types:</p>
+        <source>
+&lt;module name=&quot;FileTabCharacter&quot;&gt;
+  &lt;property name=&quot;fileExtensions&quot; value=&quot;java, xml&quot;/&gt;
+&lt;/module&gt;
+        </source>
+        <p>Example:</p>
+        <source>
+//a java file
+public class test {
+  int a;                           // violation : indented using tab, no further checks
+
+  public void foo (int arg) {
+      a = arg;
+  }
+}
+
+//an XML file
+&lt;?xml version="1.0" encoding="UTF-8" ?&gt;
+&lt;UserAccount&gt;
+  &lt;FirstName&gt;John&lt;/FirstName&gt;      // violation : indented using tab, no further checks
+  &lt;LastName&gt;Doe&lt;/LastName&gt;
+&lt;/UserAccount&gt;
+
+//an HTML file
+&lt;head&gt;
+  &lt;title&gt;Page Title&lt;/title&gt;        // no check performed, file extension not specified
```

Comment:
please each file in separate source secrtion make new paragrath above source and define name of fie "Test.java", "Text.xml" "Test.html".

> .... file extension not specified

html file extension not specified in Check config

Reply:
Done

Reply:
On seperating the three source files, `mvn clean verify` gives this error for `Test.xml`:
```
[ERROR] Failures: 
[ERROR]   XdocsPagesTest.testAllXmlExamples:377 config_whitespace.xml has invalid xml (The processing instruction target matching "[xX][mM][lL]" is not allowed.): <?xml version="1.0" encoding="UTF-8" ?>
<UserAccount>
  <FirstName>John</FirstName>      // violation : indented using tab, no further checks
  <LastName>Doe</LastName>
</UserAccount>

```

So I thought of removing the first line `<?xml version="1.0" encoding="UTF-8" ?>`, but now it gives this error:
```
config_whitespace.xml has invalid Checkstyle xml (unable to parse configuration stream - Element type "UserAccount" must be declared.:5:14): 
<UserAccount>
  <FirstName>John</FirstName>      // violation : indented using tab, no further checks
  <LastName>Doe</LastName>
</UserAccount>
```

How to eliminate both issues?

Reply:
You can not put `// violation : indente ...` out of xml tags, you have to use xml comments syntax ` <!--  violation : indente ... -->

Reply:
The same for html

Reply:
Ok, I have made the changes, the `xml` string was causing issues in the `config_whitespace.xml` file (because xml parser allows only one occurence of "xml" in xml document), so I found a workaround for it, included the xml example in `<![CDATA[code-example]]>` and that worked.